code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm

```diff
@@ -94,6 +101,11 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
 		else if(!user.dropItemToGround(src))
 			to_chat(user, span_warning("You can't seem to drop \the [src]! It must be stuck to your hand somehow! Prepare for unforeseen consequences..."))
 
+	//SKYRAT EDIT ADDITION - GHOST HOTEL UPDATE
+	var/chosen_room
+	if(istype(src, /obj/item/hilbertshotel/ghostdojo)) //to don't add another one var
+		chosen_room = tgui_input_list(user, "Choose desired room:","Time to choose", hotel_maps)
```

Comment:
You need to handle the client not clicking any of the available options and just clicking X instead (which will return null in `chosen_room`).

Reply:
Done